crates/router/src/db/merchant_account.rs

```diff
@@ -257,6 +257,11 @@ impl MerchantAccountInterface for MockDb {
         merchant_key_store: &domain::MerchantKeyStore,
     ) -> CustomResult<domain::MerchantAccount, errors::StorageError> {
         let mut accounts = self.merchant_accounts.lock().await;
+        let merchant_account = domain::MerchantAccount {
+            #[allow(clippy::as_conversions)]
+            id: Some(merchant_account.id.unwrap_or(accounts.len() as i32)),
+            ..merchant_account
+        };
```

Comment:
Make `merchant_account` mutable in the function parameters, then you can use `get_or_insert()` to set the `id` field if not already set.

```rust
#[allow(clippy::as_conversions)]
merchant_account.id.get_or_insert(accounts.len() as i32);
```